modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -1304,6 +1304,142 @@ public void testAddObjectEntryWithDefaultLanguageId() throws Exception {
 				ObjectDefinitionConstants.SCOPE_SITE));
 	}
 
+	@Test
+	public void testAddObjectEntryWithDefaultValue() throws Exception {
+		ObjectDefinition objectDefinition = _createObjectDefinition(
+			Arrays.asList(
+				new BooleanObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).localized(
+					true
+				).name(
+					"localizedBooleanObjectFieldName"
+				).build(),
+				new PicklistObjectFieldBuilder(
+				).indexed(
+					true
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).listTypeDefinitionId(
+					listTypeDefinition.getListTypeDefinitionId()
+				).localized(
+					true
+				).name(
+					"localizedPicklistObjectFieldName"
+				).build()));
+
+		ObjectEntry objectEntry = _defaultObjectEntryManager.addObjectEntry(
+			dtoConverterContext, objectDefinition,
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						"localizedBooleanObjectFieldName", () -> null
+					).put(
+						"localizedPicklistObjectFieldName", () -> null
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		Map<String, Object> properties = objectEntry.getProperties();
+
+		Assert.assertNull(properties.get("localizedBooleanObjectFieldName"));
+		Assert.assertNull(properties.get("localizedPicklistObjectFieldName"));
+
+		Map<String, Object> localizedBooleanObjectFieldValue =
+			HashMapBuilder.<String, Object>put(
+				"en_US", true
+			).put(
+				"pt_BR", true
+			).build();
+
+		Map<String, Object> localizedPicklistObjectFieldValue =
+			HashMapBuilder.<String, Object>put(
+				"en_US", _listTypeEntryKey1
+			).put(
+				"pt_BR", _listTypeEntryKey2
+			).build();
+
+		_assertObjectEntryWithDefaultValue(
+			localizedBooleanObjectFieldValue, localizedPicklistObjectFieldValue,
+			_defaultObjectEntryManager.addObjectEntry(
+				dtoConverterContext, objectDefinition,
+				new ObjectEntry() {
+					{
+						properties = HashMapBuilder.<String, Object>put(
+							"localizedBooleanObjectFieldName_i18n",
+							localizedBooleanObjectFieldValue
+						).put(
+							"localizedPicklistObjectFieldName_i18n",
+							localizedPicklistObjectFieldValue
+						).build();
+					}
+				},
+				ObjectDefinitionConstants.SCOPE_COMPANY));
```

Comment:
@pedro-oliveira446 can we simplify by using `assertEquals` and `_addObjectEntry` here:

```java
Map<String, Object> localizedObjectFieldI18nValues =
	HashMapBuilder.<String, Object>put(
		"localizedBooleanObjectFieldName_i18n",
		HashMapBuilder.<String, Object>put(
			"en_US", RandomTestUtil.randomBoolean()
		).put(
			"pt_BR", RandomTestUtil.randomBoolean()
		).build()
	).put(
		"localizedPicklistObjectFieldName_i18n",
		HashMapBuilder.<String, Object>put(
			"en_US", _listTypeEntryKey1
		).put(
			"pt_BR", _listTypeEntryKey2
		).build()
	).build();

assertEquals(
	_addObjectEntry(objectDefinition, localizedObjectFieldI18nValues),
	new ObjectEntry() {
		{
			properties = localizedObjectFieldI18nValues;
		}
	}
);
```
?